### ISSUE TITLE
Unified User Registration

### ISSUE THREAD

**Issue by a-bergevin** [sourcegraph/sourcegraph#34108]

**Comment by a-bergevin** [comment]
Unified User Registration
Scope: All visitors to Sourcegraph.com are prompted to register with basic information (see Gitlab for example)
Out of scope: registration of users on prem instances

**Comment by a-bergevin** [comment]
engineering is doing discovery on current state before we can plan next steps.
